kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -547,6 +550,115 @@ protected String createAndValidateAccountingLines(org.kuali.kfs.fp.document.Proc
         return super.createAndValidateAccountingLines(pcardDocument, transaction, docTransactionDetail);
     }
     
+    @Override
+    protected ProcurementCardTargetAccountingLine createTargetAccountingLine(ProcurementCardTransaction transaction, ProcurementCardTransactionDetail docTransactionDetail) {
+        ProcurementCardTargetAccountingLine targetLine = new ProcurementCardTargetAccountingLine();
+        ProcurementCardDefault procurementCardHolderDetail = getProcurementCardDefault(transaction.getTransactionCreditCardNumber());
+        
+        targetLine.setDocumentNumber(docTransactionDetail.getDocumentNumber());
+        targetLine.setFinancialDocumentTransactionLineNumber(docTransactionDetail.getFinancialDocumentTransactionLineNumber());
+        //set error account info as default
+        targetLine.setChartOfAccountsCode(getErrorChartCode());
+        targetLine.setAccountNumber(getErrorAccountNumber());
+        targetLine.setFinancialObjectCode(getDefaultObjectCode());
+        if (hasReconciler(procurementCardHolderDetail)) {
+            targetLine.setChartOfAccountsCode(StringUtils.isNotEmpty(procurementCardHolderDetail.getChartOfAccountsCode()) ? procurementCardHolderDetail.getChartOfAccountsCode() : getErrorChartCode());
+            targetLine.setAccountNumber(StringUtils.isNotEmpty(procurementCardHolderDetail.getAccountNumber()) ? procurementCardHolderDetail.getAccountNumber() : getErrorAccountNumber());
+            targetLine.setFinancialObjectCode(StringUtils.isNotEmpty(procurementCardHolderDetail.getFinancialObjectCode()) ? procurementCardHolderDetail.getFinancialObjectCode() : getDefaultObjectCode());
+        }
+        if (ObjectUtils.isNotNull(procurementCardHolderDetail)) {
+            targetLine.setSubAccountNumber(procurementCardHolderDetail.getSubAccountNumber());
+            targetLine.setFinancialSubObjectCode(procurementCardHolderDetail.getFinancialSubObjectCode());
+        }
+        targetLine.setProjectCode(transaction.getProjectCode());
+
+        if (KFSConstants.GL_CREDIT_CODE.equals(transaction.getTransactionDebitCreditCode())) {
+            targetLine.setAmount(transaction.getFinancialDocumentTotalAmount().negated());
+        }
+        else {
+            targetLine.setAmount(transaction.getFinancialDocumentTotalAmount());
+        }
+
+        return targetLine;
+    }
+    
+    @Override
+    protected String validateTargetAccountingLine(ProcurementCardTargetAccountingLine targetLine) {
+        String errorText = KFSConstants.EMPTY_STRING;
+
+        targetLine.refresh();
+        final String lineNumber = targetLine.getSequenceNumber() == null ? KFSPropertyConstants.NEW : targetLine.getSequenceNumber().toString();
+
+        if (!accountingLineRuleUtil.isValidChart(KFSConstants.EMPTY_STRING, targetLine.getChart(), dataDictionaryService.getDataDictionary())) {
```

Comment:
Normally I would call out the lack of accessor method for injected services, but I see the kuali super lacks them as well. Instead, I'll take it as opportunity to point out this is a bad practice put on us by the code base -- don't do it this way when we can help it. (No changes requested).